src/main/java/io/github/moulberry/notenoughupdates/listener/ItemTooltipRngListener.java

```diff
@@ -0,0 +1,306 @@
+						if (!NotEnoughUpdates.INSTANCE.config.tooltipTweaks.rngMeterRunsNeeded) {
+							String name = Utils.getOpenChestName().contains("Catacombs") ? "Score" : "XP";
+							String formatCoinsPer = getFormatCoinsPer(event.itemStack, needed, 1, name);
+							if (formatCoinsPer != null) {
+								newToolTip.add(line);
+								newToolTip.add(formatCoinsPer);
+								continue;
+							}
+						}
+					}
+				}
+				nextLineProgress = false;
+			}
+
+			if (line.contains("Progress:")) {
+				nextLineProgress = true;
+			}
+			newToolTip.add(line);
+		}
+
+		event.toolTip.clear();
+		event.toolTip.addAll(newToolTip);
+	}
+
+	private String getFormatCoinsPer(ItemStack stack, int needed, int multiplier, String name) {
+		String internalName = neu.manager.getInternalNameForItem(stack);
+		double bin = neu.manager.auctionManager.getBazaarOrBin(internalName);
+		if (bin < 0) return null;
+
+		double coinsPer = (bin / needed) * multiplier;
+		String format = StringUtils.shortNumberFormat(coinsPer);
+		return "§7Coins per " + name + ": §6" + format + " coins";
+	}
+
+	private void fractionDisplay(List<String> newToolTip, String line) {
+		boolean shift = Keyboard.isKeyDown(Keyboard.KEY_LSHIFT) || Keyboard.isKeyDown(Keyboard.KEY_RSHIFT);
+		if (!pressedShiftLast && shift) {
+			showSlayerRngFractures = !showSlayerRngFractures;
+		}
+		pressedShiftLast = shift;
+
+		String[] split = line.split(Pattern.quote(" §7("));
+		String start = split[0];
+		String string = StringUtils.cleanColour(split[1]);
+		string = string.replace("%", "").replace(")", "");
+
+		String addText;
+		int baseChance;
+		if (!string.contains(" ")) {
+			baseChance = calculateChance(string);
+			String format = GuiProfileViewer.numberFormat.format(baseChance);
+			addText = " §7(1/" + format + ")";
+		} else {
+			split = string.split(" ");
+			baseChance = calculateChance(split[0]);
+			String base = GuiProfileViewer.numberFormat.format(baseChance);
+			String increased = GuiProfileViewer.numberFormat.format(calculateChance(split[1]));
+			addText = " §7(§8§m1/" + base + "§r §71/" + increased + ")";
+		}
+
+		if (showSlayerRngFractures) {
+			newToolTip.add(start + addText);
+			newToolTip.add("§8[Press SHIFT to show odds as percentages]");
+		} else {
+			newToolTip.add(line);
+			newToolTip.add("§8[Press SHIFT to show odds as fractures]");
+		}
+	}
+
+	/**
+	 * This adds support for the /neureloadrepo command
+	 */
+	@SubscribeEvent(priority = EventPriority.LOWEST)
+	public void onRepoReload(RepositoryReloadEvent event) {
+		repoReloadNeeded = true;
+	}
+
+	public void checkUpdateData() {
+		if (repoReloadNeeded) {
+			updateRepoData();
+		}
+	}
+
+	private void updateRepoData() {
+		slayerData.clear();
+		dungeonData.clear();
+
+		JsonObject leveling = Constants.LEVELING;
+		if (!leveling.has("slayer_boss_xp") ||
+			!leveling.has("slayer_highest_tier") ||
+			!leveling.has("slayer_tier_colors") ||
+			!leveling.has("rng_meter_dungeon_score")) {
+			return;
```

Comment:
Not sure if you want an `Utils.showOutdatedRepoNotification()` here. It's a thing we have, but using it is a stylistic choice ill leave to you.

Reply:
Done